### PR TITLE
CLOUDP-82915: Add Readinessprobe Dockerfile

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

This PR adds a simple Dockerfile that builds the readiness probe. The enterprise codebase will use this image.

Note: This is still not being used in the community code base, refactoring the code to use the new readiness probe will come in a future PR.

### CODE REVIEW

File: scripts/dev/dockerfiles/Dockerfile.readiness [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM golang:1.13-alpine as builder
```
Comment (irajdeep) [inline scripts/dev/dockerfiles/Dockerfile.readiness:1]: should we bump the go version here as well?

File: scripts/dev/dockerfiles/Dockerfile.readiness [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+COPY ./cmd/readiness /build/
+COPY ./pkg /build/pkg
+COPY ./api /build/api
+COPY ./go.mod /build/
```
Comment (irajdeep) [inline scripts/dev/dockerfiles/Dockerfile.readiness:6]: So we have different go.mod file for the probe, can/should we use the project's go.mod?

File: scripts/dev/dockerfiles/Dockerfile.readiness (hunk 1 above)
Comment (chatton) [reply scripts/dev/dockerfiles/Dockerfile.readiness:1]: yes I think there's no harm doing that

File: scripts/dev/dockerfiles/Dockerfile.readiness (hunk 2 above)
Comment (chatton) [reply scripts/dev/dockerfiles/Dockerfile.readiness:6]: this is a single go.mod file, we just have a different main (in this case `cmd/readiness/main.go`)

**Review by irajdeep — APPROVED** [review]
LGTM 👍

**Pull request merged**